### ISSUE TITLE
error: name 'plt' is not defined - Matplotlib is a hard dependency?

### ISSUE THREAD

**Issue by samhatchett** [USEPA/WNTR#35]
When Matplotlib is not installed, this error is generated upon import. According to the [docs](http://wntr.readthedocs.io/en/latest/installation.html#requirements), Matplotlib is an _optional_ dependency. However, I can find no way to use the WNTR package without installing Matplotlib.

I also see this when using the continuumio/anaconda3:5.0.1 docker image, which does have matplotlib pre-installed. So apparently there's some compatibility problems there as well.

```
Traceback (most recent call last):
  File "./epanet_worker.py", line 1, in <module>
    from wntr import network
  File "/opt/conda/lib/python3.6/site-packages/wntr/__init__.py", line 6, in <module>
    from wntr import graphics
  File "/opt/conda/lib/python3.6/site-packages/wntr/graphics/__init__.py", line 4, in <module>
    from wntr.graphics.network import plot_network, plot_interactive_network
  File "/opt/conda/lib/python3.6/site-packages/wntr/graphics/network.py", line 20, in <module>
    node_size=20, node_range = [None,None], node_cmap=plt.cm.jet, node_labels=False,
NameError: name 'plt' is not defined
```

**Comment by kaklise** [comment]
Matplotlib should be a optional dependency and we have that issue fixed on a different branch.  I'll get that update pushed to USEPA/master as soon as possible. Thanks!

**Comment by samhatchett** [comment]
in addition to it being documented as an optional dependency, I've also found that pyplot doesn't import in the 64-bit Python interpreter. No issues using 32-bit - so that was also a complicating factor in the above. thanks!

**Comment by kaklise** [comment]
We are using 64-bit Python without issue.  Travis CI also tests against 64-bit Python.  It might be an issue with the matplotlib installation.  Anaconda installers are generally seemless, but sometimes issues crop up.  Travis is currently testing against Matlab 2.1.1, but we have not had issues with older versions of Matplotlib (going back to 1.5.1).

**Comment by dbhart** [comment]
I'm sure he's using a Windows machine in this case - no one had 32bit Linux boxes or Macs

Sent from VMware Boxer (Android)

On Jan 19, 2018 12:11, Katherine Klise <notifications@github.com> wrote:

We are using 64-bit Python without issue. Travis CI also tests against 64-bit Python. It might be an issue with the matplotlib installation. Anaconda installers are generally seemless, but sometimes issues crop up. Travis is currently testing against Matlab 2.1.1, but we have not had issues with older versions of Matplotlib (going back to 1.5.1).

-

**Comment by kaklise** [comment]
That should not be a problem.  I'm working on a windows machine.

**Comment by dbhart** [comment]
I have tested on 64-bit Python on Mac and we test travis on 64-bit Python on Linux and kaklise uses 64-bit windows Python. That said, it may depend on what version of conda and/or base python mixture you are using. Can you tell us that info?

**Comment by kaklise** [comment]
PR #36 removes the hard dependency on matplotlib.  We'll still want to figure out the issue above so you can generate graphics.

**Comment by samhatchett** [comment]
I am on a Mac. sorry for leaving that part out.

**Comment by samhatchett** [comment]
sorry - I'm on a Mac. matplotlib v2.0.2

**Comment by dbhart** [comment]
I’m on a mac with Anaconda Python27 and Python36 64-bit / matplotlib 1.5.3, but let me see if I can upgrade my matplotlib and see if it breaks

**Comment by michaelbynum** [comment]
I have had problems installing matplotlib with pip on a Mac. However, it
works fine if I install with Conda.

Michael

**Comment by samhatchett** [comment]
I may have some configuration issues on my machine, but it wouldn't explain the weirdness with the Anaconda3 docker image. In any case, that's not the primary concern of this project.

sorry to stir up trouble here related to other-party packages. the main point of the issue is that matplotlib was a hard dependency for using wntr, which disagreed with the docs. 

I was trying to run WNTR in a dockerized environment, and having some trouble picking a base image. Finally settled on the Python3 base, installing corequisites via pip.

But for those interested:

```
$ docker run -it continuumio/anaconda3:latest python -c '\
import matplotlib; \
print(matplotlib.__version__); \
print(matplotlib.pyplot) \
'

2.1.0
Traceback (most recent call last):
  File "<string>", line 4, in <module>
AttributeError: module 'matplotlib' has no attribute 'pyplot'
```

... which is probably unexpected.

**Comment by michaelbynum** [comment]
No apologies necessary. We are happy to (at least try) to help. We appreciate you bringing the matplotlib dependency issue to our attention.

**Comment by dbhart** [comment]
Definitely no apologies needed, and I may try out the docker image anyway, because I have a different project where they may want docker down the line, and it does have a hard dependency on matplotlib, so it would be good to double check. Thanks!

Sent from VMware Boxer (Android)

**Comment by kaklise** [comment]
closed by PR #36.

**Comment by purplemaestro** [comment]
### dad
